lib/App/Rakubrew/Shell/Fish.pm

```diff
@@ -118,7 +118,15 @@ sub completions {
     my $self = shift;
     my @words = @_;
 
-    my @completions = $self->get_completions(@words - 1, @words);
+    my $index = scalar(@words);
+    # Strip command name.
+    while (@words > 0) {
+        my $word = shift @words;
+        $index--;
+        last if $word =~ /(^|\W)$brew_name$/;
```

Comment:
I really, really can't remember why I put this complex piece of code into the Zsh shell hook. From all I know a simple `shift @words` should suffice to remove the command name. (That's how it's done in all the other shell hooks.)

Reasoning: Autocompletion only works when using the shell hook. When using the hook rakubrew will be called by its short name and not with its full path, otherwise the hook would break.

So if you also don't know any reason, can you just replace this with `shift @words;` ?

Reply:
I would still keep it there just to be on the safe side. Move it into a helper method on `App::Rakubrew::Shell` perhaps and call from both fish and zsh hooks then. The point of having it there could be, for example, if in a case like `ENV_VAR=42 rakubrew` we get both the assignment and `rakubrew` as arguments. So far, it's not the case for fish, but I wouldn't bet for other shells or for other combinations of commands.

Reply:
If we keep it for the safe side, then we should do it in all the different shell hooks. A unification in App::Rakubrew::Shell makes sense. Maybe even just enhance get_completions to do it.

Reply:
At first I though that get_completions must not care about this aspect. But then it made more sense as I looked at it from another angle.

Reply:
Due to PowerShell peculiarities, I decide to keep the logic outside of `get_completions()`. After all, it anyway seems to be the right thing now, even though it adds a little to hooks boilerplate.